### PR TITLE
Remove need for loader and boot the application directly even when hot reloading

### LINKED ISSUE #6508: Hot reload "syncing files to device" takes a long time
Can we maybe compress the files or find a quicker way to send them or send them with the initial package or something?

### PR DESCRIPTION
**Pull request by johnmccutchan**
- [x] Remove the two stage initial boot process.
- [x] Remove the loader screen app.
- [x] Don't report reload times for the initial reload because we are
switching from a snapshot to source and that will always be worst case.

Fixes https://github.com/flutter/flutter/issues/6508

### CODE REVIEW

**Comment by johnmccutchan** [comment]
@Hixie @devoncarew @tvolkert

File: packages/flutter_tools/lib/src/android/android_device.dart [hunk 1]
```diff
@@ -352,7 +352,7 @@ class AndroidDevice extends Device {
           Future<List<int>> scrapeServicePorts = Future.wait(
             <Future<int>>[observatoryDiscovery.nextPort(), diagnosticDiscovery.nextPort()]
           );
-          List<int> devicePorts = await scrapeServicePorts.timeout(new Duration(seconds: 20));
+          List<int> devicePorts = await scrapeServicePorts.timeout(new Duration(seconds: 500));
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/android/android_device.dart]: Debugging change mistakenly pushed?

File: packages/flutter_tools/lib/src/hot.dart [hunk 2]
```diff
@@ -244,7 +236,7 @@ class HotRunner extends ResidentRunner {
     LaunchResult result = await futureResult;
 
     if (!result.started) {
-      printError('Error launching DevFS loader on ${device.name}.');
+      printError('Error launching applicaiton on ${device.name}.');
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/hot.dart]: sp: application

**Review by tvolkert — COMMENTED** [review]
The Travis failures look unrelated - maybe kick the bot to try again?

**Comment by devoncarew** [comment]
lgtm, but it looks like travis is unhappy

File: packages/flutter_tools/lib/src/hot.dart (hunk 2 above)
Comment (johnmccutchan) [inline packages/flutter_tools/lib/src/hot.dart]: Done.

File: packages/flutter_tools/lib/src/android/android_device.dart (hunk 1 above)
Comment (johnmccutchan) [inline packages/flutter_tools/lib/src/android/android_device.dart]: Yes and fixed.

**Comment by tvolkert** [comment]
LGTM

**Comment by johnmccutchan** [comment]
I need to wait until we have a green engine revision.

**Pull request merged**